### PR TITLE
589 handle subauthorities

### LINKED ISSUE #589: Ensure QA lookups refer to subauthorities and make correct call
Right now the InputLookupQA component uses the following call structure: "client.apis.SearchQuery .GET_searchAuthority" and passes the authority value in the subauthority field.  When calling the subauthority, the code should use "GET_searchSubauthority" instead.  The apidoc currently uses a space in the tag "Search Query", which cannot be converted to the format for "client.apis.SearchQuery" (spaces don't appear to work).  I'll be updating the apidoc to replace spaces in the tags.  Although this will affect how the documentation looks (since the tags are used as headings), this seems the appropriate method to use to get this code to work.  I'll be making a related issue on QA server to suggest updates to the apidoc there.

### PR DESCRIPTION
**Pull request by hudajkhan**
Fixes #589 

Updated apidoc to enable calling all methods (spaces needed to be removed from tags) using current pattern for calls.  If subauthority present, subauthority API call is made. Otherwise authority API is used.

### CODE REVIEW

File: src/components/editor/InputLookupQA.jsx [hunk 1]
```diff
@@ -118,21 +119,43 @@ class InputLookupQA extends Component {
...
-                                })
-                                .catch(function(err) {
-                                  console.error("Error in executing lookup against source", err)
-                                  //return information along with the error in its own object
-                                  return { isError: true, errorObject: err }
-                                })
+                              //Subauthorities require a different API call than authorities so need to check if subauthority is available
+                              //The only difference between this call and the next one is the call to Get_searchSubauthority instead of 
+                              //Get_searchauthority.  If there is a way to somehow pass that in a variable name/dynamically, that would be better
+                              if(subauthority) {
```
Comment (mjgiarlo) [inline src/components/editor/InputLookupQA.jsx]: @hudajkhan Re: your comment, how about:

```javascript
const actionFunction = subauthority === undefined ? client.apis.SearchQuery.GET_searchAuthority : client.apis.SearchQuery.GET_searchSubauthority

return actionFunction({
  q: query,
  vocab: authority,
  subauthority: subauthority,
  maxRecords: maxRecords,
  lang: language
})
// ...
```

File: src/components/editor/InputLookupQA.jsx [hunk 2]
```diff
@@ -84,6 +84,7 @@ class InputLookupQA extends Component {
   render() {
     let authority, subauthority, language
     const lookupConfigs = this.props.lookupConfig
+    const maxRecords = 8
```
Comment (mjgiarlo) [inline src/components/editor/InputLookupQA.jsx]: Can you add this to `src/Config.js` as `maxRecordsForQALookups`?

Comment (hudajkhan) [reply src/components/editor/InputLookupQA.jsx]: Like this? 
static get maxRecordsForQALookups() {
	  return 8
 }

Comment (mjgiarlo) [reply src/components/editor/InputLookupQA.jsx]: @hudajkhan 	How about:

```javascript
static get maxRecordsForQALookups() {
  return process.env.MAX_RECORDS_FOR_QA_LOOKUPS || 8
}
```

This lets us configure the default via environment variables. Thanks!

**Comment by mjgiarlo** [comment]
@hudajkhan 	Can you also amend the test suites for `InputLookupQA` and `Config` (once changed) to reflect the changes you've made?

File: src/components/editor/InputLookupQA.jsx (hunk 1 above)
Comment (hudajkhan) [reply src/components/editor/InputLookupQA.jsx]: followed this pattern and now in code (tested and works)

File: src/components/editor/InputLookupQA.jsx (hunk 2 above)
Comment (hudajkhan) [reply src/components/editor/InputLookupQA.jsx]: Added this to Config.js

File: src/components/editor/InputLookupQA.jsx (hunk 1 above)
Comment (hudajkhan) [reply src/components/editor/InputLookupQA.jsx]: I'm going to resolve this conversation.  I'm taking the initiative or something but feel free to unresolve if it is not... resolved.

File: src/components/editor/InputLookupQA.jsx (hunk 2 above)
Comment (hudajkhan) [reply src/components/editor/InputLookupQA.jsx]: Also resolving because I followed instructions

**Comment by hudajkhan** [comment]
Added a test for Config.  Mulling over test for input qa.  The changes to inputLookupQA were around identifying if a subauthority is passed from lookup config and then calling the right method from the apidoc.

File: src/components/editor/InputLookupQA.jsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { connect } from 'react-redux'
 import { getProperty } from '../../reducers/index'
 import { changeSelections } from '../../actions/index'
 import { booleanPropertyFromTemplate, defaultValuesFromPropertyTemplate } from '../../Utilities'
+import Config from '../../../src/Config'
```
Comment (mjgiarlo) [inline src/components/editor/InputLookupQA.jsx]: This component is already in the `src/` directory, so I think the path should be `'../../Config'`

File: src/Config.js [hunk 4]
```diff
@@ -63,6 +63,10 @@ class Config {
   static get cognitoTestUserPass() {
     return process.env.COGNITO_TEST_USER_PASS
   }
+  
+  static get maxRecordsForQALookups() {
+	  return process.env.MAX_RECORDS_FOR_QA_LOOKUPS || 8
```
Comment (mjgiarlo) [inline src/Config.js]: Awesome, thanks!

File: src/components/editor/InputLookupQA.jsx (hunk 1 above)
Comment (mjgiarlo) [reply src/components/editor/InputLookupQA.jsx]: awesome!

**Comment by hudajkhan** [comment]
Having an issue with the test I intended to evaluate the function for selecting the API.  I'm commenting it out for now until I figure out how to do this properly (I'm also running into interesting challenges on being able to run tests on any of my local environments but that is a different story).  thanks @jgreben for helping me step through some of it!

**Comment by hudajkhan** [comment]
Down to 0 errors and 12 warnings but the cutoff is 11 warnings.

**Comment by mjgiarlo** [comment]
Hi, @hudajkhan! Yesterday, I believe @jermnelson said that your Sinopia time was running low. If you don't have time to knock out the last warning, I'd be happy to pick up this branch, work on the build, rebase on `master`, and help finish. How would you like to proceed?

**Comment by hudajkhan** [comment]
Hi @mjgiarlo ! If you could help me understand how to knock out that last warning (and maybe how to properly write the test), that would be great.   I can then update the code.  (fyi I slacked you too)

**Pull request merged**